contracts/Distributor.sol

```diff
@@ -0,0 +1,259 @@
+    uint256 private _totalETHShares;
+    uint256 private _totalETHDistributed;
+
+
+     /*///////////////////////////////////////////////////////////////
+                            Mappings
+    //////////////////////////////////////////////////////////////*/
+   
+
+    mapping(uint256 => address) private _payeesForETH;
+    // Specification of total share for ETHER per wallet
+    mapping(address => uint256) private _sharesForETH;
+    // Amount of ETH paid out to the account
+    mapping(address => uint256) private _distributedETH;
+
+    mapping(IERC20 => uint256) private _numRegisteredForToken; // Counter for each ERC20 Token to track no. of registered payees
+    mapping(IERC20 => mapping(uint256 => address)) private _payeesForToken; // IERC20 token -> index -> registered payee
+
+    mapping(IERC20 => uint256) private _totalTokenShares;
+    mapping(IERC20 => uint256) private _totalTokenDistributed;
+    // Specification of total shares tagged to the ERC20 Token Contract Address per wallet
+    mapping(IERC20 => mapping(address => uint256)) private _sharesForTokens; // IERC20 token -> account -> share
+    // Amount of Token specified paid out to the account
+    mapping(IERC20 => mapping(address => uint256)) private _distributedTokens; // IERC20 token -> account -> distributed Tokens
+
+
+         /*///////////////////////////////////////////////////////////////
+                            Modifiers
+    //////////////////////////////////////////////////////////////*/
+
+    modifier nonZeroAddress(address account_) {
+        require(account_ != address(0), "Distributor: Null address not allowed.");
+        _;
+    }
+
+    modifier validShares(uint256 shares_) {
+         require(shares_ > 0, "Distributor: Invalid shares specified.");
+         _;
+    }
+
+        /*///////////////////////////////////////////////////////////////
+                            Events
+    //////////////////////////////////////////////////////////////*/
+    event RegisterPayeeForETH(address account, uint256 shares);
+    event RegisterPayeeForToken(address account, IERC20 token, uint256 shares );
+ 
+    event ETHFundClaimed(address account,uint256 shares);
+    event TokenFundClaimed(address account, IERC20 token, uint256 shares);
+
+    event ETHReceived(address from, uint256 amount);
+    event TokenReceived(address from, IERC20 token, uint256 amount);
+
+
+           /*///////////////////////////////////////////////////////////////
+                            Constructor
+    //////////////////////////////////////////////////////////////*/
+    // @Note: SHare 
+    constructor(address[] memory accounts_, uint256[] memory shares_) payable {
+        require(accounts_.length == shares_.length, "Distributor: Mismatch of addresses and shares.");
+        for(uint256 i = 0; i < accounts_.length; i++ ){
+            _registerPayeesForETH(accounts_[i], shares_[i]);
```

Comment:
I think not exposing this function to public makes it that if the owner want to airdrop for some users in the future, he won't be able to

Reply:
Hence, strongly recommend to move it out of the constructor & expose it to be a public/external function

Reply:
Yeah actually should make it flexible. Thanks a lot and will make the necessary amends!